src/Test/QuickCheck/Monoids.hs

```diff
@@ -0,0 +1,54 @@
+{-# LANGUAGE ExistentialQuantification #-}
+
+module Test.QuickCheck.Monoids
+  ( All (..)
+  , Any (..)
+  ) where
+
+import Data.List.NonEmpty as NonEmpty
+import Data.Semigroup (Semigroup (..))
+import Test.QuickCheck.Property
+
+-- | Conjunction monoid build with `.&&.`.
+--
+-- Use `property @All` as an accessor which doesn't leak
+-- existential variables.
+--
+-- Note: monoid laws are satisfied up to `isSuccess` unless one is using
+-- `checkCoverage`.
+--
+data All = forall p. Testable p => All { getAll :: p }
+
+instance Testable All where
+    property (All p) = property p
+
+instance Semigroup All where
+    All p <> All p' = All (p .&&. p')
+    sconcat = All . conjoin . NonEmpty.toList
+
+instance Monoid All where
+    mempty = All True
+    mconcat = All . conjoin
+
+
+-- | Disjunction monoid build with `.||.`.
+--
+-- Use `property @Any` as an accessor which doesn't leak
+-- existential variables.
```

Comment:
Consider you have an abient function which returns `Property`, if you try to use `getAny` GHC will complain that `forall p. Testable p => p` is not `Property`, or it might error by saying that an existential `p` is leaking, in that scenario just using `property` (or `property . getAny`) is what one needs.

```
λ a  = All True
λ :t a
a :: All
λ :t getAll a

<interactive>:1:1: error: [GHC-55876]
    • Cannot use record selector ‘getAll’ as a function due to escaped type variables
    • In the expression: getAll a
    Suggested fix: Use pattern-matching syntax instead
    
λ :t case a of All x -> x
<interactive>:1:20: error: [GHC-25897]
    • Couldn't match expected type ‘p’ with actual type ‘p1’
      ‘p1’ is a rigid type variable bound by
        a pattern with constructor:
          All :: forall p. Testable p => p -> All,
        in a case alternative
        at <interactive>:1:11-15
      ‘p’ is a rigid type variable bound by
        the inferred type of it :: p
        at <interactive>:1:1
    • In the expression: x
      In a case alternative: All x -> x
      In the expression: case a of All x -> x
    • Relevant bindings include x :: p1 (bound at <interactive>:1:15)
```
but
```
λ :t property a
property a :: Property
```

Reply:
Ok, I see what you mean. I'm inclined to say at this point that it's better to just not say anything / point out that the types are  `Testable`.

Also, the type of `getEvery` and `getSome` are such that they can basically never be used. I'd prefer to remove them from the code all together to keep it cleaner.